tensorflow/lite/micro/kernels/xcore/xcore_dispatcher.h

```diff
@@ -73,42 +60,93 @@ constexpr size_t kBytesPerStackword = 4;
 constexpr size_t kWordAlignment = 4;
 constexpr size_t kDoubleWordAlignment = 8;
 
-typedef struct TaskArray {
-  ATTRIBUTE_THREAD_FUNCTION thread_function_t function;
-  size_t stack_size;
-  char* stack;
-  int size;
-  void* arguments[kMaxThreads];
-} TaskArray;
+typedef void (*thread_function_t)(void *);
+typedef tflite::ops::micro::xcore::PersistentArray<void *>
+    dispatcher_arg_array_t;
```

Comment:
We should stick to the idiomatic C++ way of doing it (and also follow the naming conventions):
```suggestion
using ThreadFunction = void (*)(void *);
using DispatcherArgArray = tflite::ops::micro::xcore::PersistentArray<void *>;
```